Nef_3/include/CGAL/Nef_3/SNC_point_locator.h

```diff
@@ -849,7 +849,24 @@ class SNC_point_locator_by_spatial_subdivision :
           _CGAL_NEF_TRACEN("found on facet...");
           return make_object(f);
         }
-        if( is.does_intersect_internally(s,f,ip)) {
+
+        // We next check if v is a vertex on the face to avoid a geometric test
+        bool v_vertex_of_f = false;
+        Halffacet_cycle_iterator fci;
+        for(fci=f->facet_cycles_begin(); (! v_vertex_of_f) && (fci!=f->facet_cycles_end()); ++fci) {
+          if(fci.is_shalfedge()) {
+            SHalfedge_around_facet_circulator sfc(fci), send(sfc);
+            CGAL_For_all(sfc,send) {
+              if(sfc->source()->center_vertex() ==  v){
+                v_vertex_of_f = true;
+                break;
+              }
+            }
+          }
+        }
+
+
+        if( (! v_vertex_of_f) &&  is.does_intersect_internally(s,f,ip) ) {
```

Comment:
The fact that `break` cannot exit more than one loop should give an hint that there should be a function there: `return` can exit any block of code!

```suggestion
        // We next check if v is a vertex on the face to avoid a geometric test
        auto v_vertex_of_f = [&]() {
          for(auto fci=f->facet_cycles_begin(); fci!=f->facet_cycles_end(); ++fci) {
            if(fci.is_shalfedge()) {
              SHalfedge_around_facet_circulator sfc(fci), send(sfc);
              CGAL_For_all(sfc,send) {
                if(sfc->source()->center_vertex() ==  v){
                  return true;
                }
              }
            }
          }
          return false;
        };

        if( (! v_vertex_of_f()) &&  is.does_intersect_internally(s,f,ip) ) {
```

Reply:
I agree (not surprisingly :joy: )

Reply:
Is what I wrote plain false, or do you consider the function  just more elegant and modern?

Reply:
The second: what you wrote is correct. The only difference is the extra test of `v_vertex_of_f` in the outer loop. I doubt a performance profiler can see the difference.

Reply:
Some might even say that factoring into a separate function is a good way to break up long blocks of code and make the code self-documenting without the need for additional comments ;) Of course, If performance is the main concern then stylistic considerations like these may need to be ignored.

Reply:
If you don't mind I would prefer to keep it as is.